src/main/java/com/uber/cadence/serviceclient/WorkflowServiceTChannel.java

```diff
@@ -237,6 +284,11 @@ private static String getEndpoint(String service, String method) {
     } catch (ExecutionException e) {
       throw new TException(e);
     } catch (TChannelError e) {
+      try {
+        resetSubchannelPeers();
```

Comment:
I think it's better not reset here. The latency of API will become unpredictable with this change. Maybe worker/workerFactory could have a scheduledExecutionService to do heathcheck and reset endpont every 10s.

Reply:
Ok. We added it everywhere to try to reduce the amount of 'downtime' a client would be unable to talk to the cluster

Reply:
I agreed it would work in most times. But still not a good practice to do for this API. 
By contract, each service API in this component should do only one thing and sending request to the service. 

Having a background process to reset should be good enough. If 10s is too long for downtime, make it configurable and even default to 1 second. Heathcheck for every 1s is fine. 

We should add some metrics to this background process for visibility. In case we see a worker constatnly resetting endpoint, there is something wrong elsewhere. 

Not only additional latency, this `TChannelError` may not be safe for resetting endpoint for all service APIs, which may end up resetting for every request.